### PR TITLE
Preset easings

### PR DESCRIPTION
**Pull request by AriaMinaei**
This (draft) PR will implement pre-defined easings, as requested in #28.

The idea is that we re-use the curve editor's string input so it recognizes a string like `Quad In Out` (or fuzzy) `qio` and replaces it with `0.455, 0.030, 0.515, 0.955` as per [this line](https://github.com/AriaMinaei/theatre/blob/c5b82dfbe18a86a976f756d31ca127ba1f3fee1e/theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/KeyframeEditor/CurveEditorPopover/CurveEditorPopover.tsx#L30).

So, imagine an autocomplete list showing up underneath the string input that lists all 24 pre-defined easings, which the user can scroll through, click on its items, or narrow down with fuzzy search.

https://user-images.githubusercontent.com/593118/144756484-0e963605-8fc2-4c28-95a2-0139dc4b27f1.mp4

The relevant components are [CurveEditorPopover](https://github.com/AriaMinaei/theatre/blob/c5b82dfbe18a86a976f756d31ca127ba1f3fee1e/theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/KeyframeEditor/CurveEditorPopover/CurveEditorPopover.tsx#L63) and [BasicAutocompleteInput](https://github.com/AriaMinaei/theatre/blob/c5b82dfbe18a86a976f756d31ca127ba1f3fee1e/theatre/studio/src/uiComponents/form/BasicAutocompleteInput.tsx#L54).

### CODE REVIEW

**Comment by AndrewPrifer** [comment]
Work left to do:
- make colors/spacing consistent and ideally conform some definition of a design system
- have keyboard navigation for the autocomplete items
- preview animation for the connection when hovering
- clean up the code
- remove CSS cubic-bezier format as canonical representation, make the input function more like an omnibox that can function as a search field for presets, but later could also apply different easing types/functions

File: theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/KeyframeEditor/CurveEditorPopover/CurveEditorPopover.tsx [hunk 1]
```diff
@@ -1,30 +1,116 @@
...
-const Label = styled.div`
-  ${propNameText};
-  white-space: nowrap;
+const EasingCurveContainer = styled.div`
+  display: flex;
+  padding: 6px;
+  background: rgba(255, 255, 255, 0.1);
+`
+
+const SearchBox = styled.input.attrs({type: 'text'})`
```
Comment (AriaMinaei) [inline theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/KeyframeEditor/CurveEditorPopover/CurveEditorPopover.tsx:97]: I'm getting this on FF 94:

image

File: theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/KeyframeEditor/CurveEditorPopover/CurveEditorPopover.tsx [hunk 2]
```diff
@@ -102,45 +211,134 @@ const CurveEditorPopover: React.FC<
...
+          placeholder="Search presets..."
+          onChange={(e) => {
+            setFilter(e.target.value)
+          }}
+          ref={inputRef}
+          onBlur={() => props.onRequestClose()}
+          onKeyPress={(e) => {
+            if (e.key === 'Escape') {
+              props.onRequestClose()
+            } else if (e.key === 'Enter') {
```
Comment (AriaMinaei) [inline theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/KeyframeEditor/CurveEditorPopover/CurveEditorPopover.tsx:252]: This is yet to be implemented, right?

Here I'm entering "l" for "linear" and hitting Enter, but linear isn't applied here:
https://user-images.githubusercontent.com/593118/145195837-98bbdc28-aacf-4949-8ad5-b78d48f1f905.mp4

File: theatre/studio/src/uiComponents/form/BasicAutocompleteInput.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  mode: 'editingViaKeyboard'
+  currentEditedValueInString: string
+  valueBeforeEditing: string
+}
+
+type IState = IState_NoFocus | IState_EditingViaKeyboard
+
+const alwaysValid = (v: string) => true
+
+const BasicStringInput: React.FC<{
```
Comment (AriaMinaei) [inline theatre/studio/src/uiComponents/form/BasicAutocompleteInput.tsx:41]: I guess we should rename this to `BasicAutocompleteInput`

**Review by AriaMinaei — COMMENTED** [review]
Hey @AndrewPrifer, this looks really good! Here is my first review pass.

File: theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/KeyframeEditor/CurveEditorPopover/CurveEditorPopover.tsx (hunk 2 above)
Comment (AndrewPrifer) [reply theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/KeyframeEditor/CurveEditorPopover/CurveEditorPopover.tsx:252]: Probably an accidental omission, will fix

**Comment by AriaMinaei** [comment]
I took a pass at colors and spacing here at #53.

**Comment by AriaMinaei** [comment]
> conform some definition of a design system

We don't have any... yet :D

File: theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/KeyframeEditor/CurveEditorPopover/CurveEditorPopover.tsx (hunk 1 above)
Comment (AndrewPrifer) [reply theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/KeyframeEditor/CurveEditorPopover/CurveEditorPopover.tsx:102]: Firefox doesn't let grids overflow their own element when the height is fixed so we need an extra inner div for the grid.

**Pull request merged**